lib/src/git.rs

```diff
@@ -355,9 +358,21 @@ pub fn fetch(
     fetch_options.proxy_options(proxy_options);
     let callbacks = callbacks.into_git();
     fetch_options.remote_callbacks(callbacks);
-    let refspec: &[&str] = &[];
+    if globs.iter().any(|glog| glog.contains(':')) {
+        return Err(GitFetchError::InvalidGlob);
+    }
+    let refspecs = globs
+        .iter()
+        .map(|glob| {
+            // if let Some(glob) = glob.strip_prefix('^') {
+            //     format!("^refs/heads/{glob}:^refs/heads/{glob}")
+            // } else {
+            format!("refs/heads/{glob}:refs/heads/{glob}")
+            // }
+        })
+        .collect_vec();
     tracing::debug!("remote.download");
-    remote.download(refspec, Some(&mut fetch_options))?;
+    remote.download(&refspecs.iter().collect_vec(), Some(&mut fetch_options))?;
```

Comment:
Why not just `&refspecs`? `String` implements `AsRef<Str>` and `IntoCString`.

Reply:
This seems to work. I still have no intuition for when it does.